arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -1176,11 +1032,21 @@ ResultT<TRI_voc_tick_t> SynchronizeShard::catchupWithReadLock(
 Result SynchronizeShard::catchupWithExclusiveLock(
     std::string const& ep, std::string const& database, LogicalCollection& collection,
     std::string const& clientId, std::string const& shard, std::string const& leader,
-    SyncerId const syncerId, TRI_voc_tick_t lastLogTick, VPackBuilder& builder) {
+    SyncerId const syncerId, TRI_voc_tick_t lastLogTick, 
+    std::shared_ptr<DatabaseTailingSyncer> tailingSyncer) {
+
+  TRI_ASSERT(tailingSyncer != nullptr);
+
   uint64_t lockJobId = 0;
   LOG_TOPIC("da129", DEBUG, Logger::MAINTENANCE)
       << "synchronizeOneShard: startReadLockOnLeader: " << ep << ":" << database
       << ":" << collection.name();
+  
+  // we should not yet have an upper bound for WAL tailing.
+  // the next call to startReadLockOnLeader may set it if the leader already implements
+  // it (ArangoDB 3.8.3 and higher)
```

Comment:
This should say "ArangoDB 3.7.16 and higher" for 3.7.

Reply:
Rephrased as "ArangoDB 3.7.16, 3.8.3 and higher".

Reply:
Ok, comment updated.